librz/bin/format/luac/luac_bin.c

```diff
@@ -0,0 +1,304 @@
+	rz_return_val_if_fail(proto_name || entry, NULL);
+	if (debug_name == NULL) {
+		return simple_build_upvalue_symbol(proto_name, entry);
+	}
+
+	return rz_str_newf("%s_upvalue_%s", proto_name, debug_name);
+}
+
+void _luac_build_info(LuaProto *proto, LuacBinInfo *info) {
+	/* process proto header info */
+	char *section_name;
+	char *symbol_name;
+	char *proto_name;
+	RzListIter *iter;
+
+	ut64 current_offset;
+	ut64 current_size;
+
+	int i = 0; // iter
+
+	// 0. check if stripped (proto name is lost)
+	if (proto->name_size == 0 || proto->proto_name == NULL) {
+		// replace name with current offset
+		proto_name = rz_str_newf("fcn.%08llx", proto->offset);
+	} else {
+		proto_name = rz_str_new((char *)proto->proto_name);
+	}
+
+	// 1.1 set section name as function_name.header
+	current_offset = proto->offset;
+	current_size = proto->size;
+	section_name = rz_str_newf("%s.header", proto_name);
+	luac_add_section(info->section_list, section_name, current_offset, current_size, false);
+	RZ_FREE(section_name);
+
+	// 1.2 set section name as function_name.code
+	current_offset = proto->code_offset;
+	current_size = proto->code_size;
+	section_name = rz_str_newf("%s.code", proto_name);
+	luac_add_section(info->section_list, section_name, current_offset, current_size, true);
+	RZ_FREE(section_name);
+
+	// 1.3 set const section
+	current_offset = proto->const_offset;
+	current_size = proto->const_size;
+	section_name = rz_str_newf("%s.const", proto_name);
+	luac_add_section(info->section_list, section_name, current_offset, current_size, false);
+	RZ_FREE(section_name);
+
+	// 1.4 upvalue section
+	current_offset = proto->upvalue_offset;
+	current_size = proto->upvalue_size;
+	section_name = rz_str_newf("%s.upvalues", proto_name);
+	luac_add_section(info->section_list, section_name, current_offset, current_size, false);
+	RZ_FREE(section_name);
+
+	// 1.5 inner protos section
+	current_offset = proto->inner_proto_offset;
+	current_size = proto->inner_proto_size;
+	section_name = rz_str_newf("%s.protos", proto_name);
+	luac_add_section(info->section_list, section_name, current_offset, current_size, false);
+	RZ_FREE(section_name);
+
+	// 1.6 debug section
+	current_offset = proto->debug_offset;
+	current_size = proto->debug_size;
+	section_name = rz_str_newf("%s.debug", proto_name);
+	luac_add_section(info->section_list, section_name, current_offset, current_size, false);
+	RZ_FREE(section_name);
+
+	// 2.1 parse local var info
+	LuaLocalVarEntry *local_var_entry;
+	rz_list_foreach (proto->local_var_info_entries, iter, local_var_entry) {
+		luac_add_string(
+			info->string_list,
+			(char *)local_var_entry->varname,
+			local_var_entry->offset,
+			local_var_entry->varname_len);
+	}
+
+	// 2.2 parse debug_upvalues
+	char **upvalue_names;
+	int real_upvalue_cnt;
+	LuaDbgUpvalueEntry *debug_upv_entry;
+	real_upvalue_cnt = rz_list_length(proto->upvalue_entries);
+	upvalue_names = RZ_NEWS0(char *, real_upvalue_cnt);
+	rz_return_if_fail(upvalue_names);
```

Comment:
I think here is unnecessary too, just add
```c
if (!upvalue_names) {
    return;
}
```

Reply:
Ok, I added it.